test/command_line/test_search_beam_position.py

```diff
@@ -93,68 +86,50 @@ def test_index_after_search(dials_data, run_in_tmpdir):
+    2. Run dials.search_beam_centre on the perturbated beam centre and original
     reflection table, check for expected output;
-    4. Run dials.index, using the datablock from the beam centre search,
-    and check that the expected unit cell is obtained and that the RMSDs are
-    smaller than or equal to some expected values."""
-
-    dials_data = dials_data("thaumatin_i04").listdir(sort=True)
-    g = [f.strpath for f in dials_data if f.ext == ".cbf"]
-
-    # beam centre from image headers: 205.28,210.76 mm
-    args = ["dials.import", "mosflm_beam_centre=207,212"] + g
-    print(args)
-    if os.name != "nt":
-        result = procrunner.run(args)
-        assert not result.returncode and not result.stderr
-    else:
-        # Can't run this command on Windows,
-        # as it will exceed the maximum Windows command length limits.
-        # So, instead:
-        with mock.patch.object(sys, "argv", args):
-            dials.command_line.dials_import.Script().run()
-    assert os.path.exists("imported.expt")
-
-    # spot-finding, just need a subset of the data
-    args = [
-        "dials.find_spots",
-        "imported.expt",
-        "scan_range=1,10",
-        "scan_range=531,540",
-    ]
-    print(args)
-    result = procrunner.run(args)
-    assert not result.returncode and not result.stderr
-    assert os.path.exists("strong.refl")
-
-    # actually run the beam centre search
-    search_beam_position.run(["imported.expt", "strong.refl"])
-    assert os.path.exists("optimised.expt")
-
-    # look at the results
-    experiments = load.experiment_list("imported.expt", check_format=False)
-    original_imageset = experiments.imagesets()[0]
-    optimized_experiments = load.experiment_list("optimised.expt", check_format=False)
-    detector_1 = original_imageset.get_detector()
-    detector_2 = optimized_experiments.detectors()[0]
-    shift = scitbx.matrix.col(detector_1[0].get_origin()) - scitbx.matrix.col(
-        detector_2[0].get_origin()
+    3. Run dials.index with the found beam centre and check that the expected
+    unit cell is obtained and that the RMSDs are smaller than or equal to some
+    expected values."""
+
+    insulin = dials_data("insulin_processed")
+
+    # load the original experiment and perturbate the beam centre by a small offset
+    experiments = load.experiment_list(insulin / "imported.expt", check_format=False)
+    original_origin = experiments[0].detector.hierarchy().get_origin()
+    shifted_origin = (
+        original_origin[0] - 1.3,
+        original_origin[1] + 1.5,
+        original_origin[2],
     )
-    print(shift)
+    experiments[0].detector.hierarchy().set_local_frame(
+        experiments[0].detector.hierarchy().get_fast_axis(),
+        experiments[0].detector.hierarchy().get_slow_axis(),
+        shifted_origin,
+    )
+    assert experiments[0].detector.hierarchy().get_origin() == shifted_origin
+    experiments.as_file(run_in_tmpdir / "shifted.expt")
```

Comment:
You can probably just use the code from `dials.modify_geometry` to do this. Something like:
```
from dials.command_line import modify_geometry
modify_geometry.run([insulin/"imported.expt", "mosflm_beam_centre=96,97"])
```
which should output a new file "modified.expt" with the updated beam centre.

Note: I think you'll have to pass along the `args` variable at this line:
https://github.com/dials/dials/blob/38aefdfe50dea24ceecf23999cf3e8dc794ae116/command_line/modify_geometry.py#L33

Reply:
I'm not too familiar with the geometry modification code - I'll just use what I have for now. Feel free to open a ticket for this if you think it'd be useful to follow up